src/main/java/chylex/customwindowtitle/TitleConfig.java

```diff
@@ -1,86 +1,115 @@
 package chylex.customwindowtitle;
 
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.ImmutableSet;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
+import java.io.InputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
 import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 public final class TitleConfig {
-	private static final ImmutableMap<String, String> DEFAULTS = ImmutableMap.<String, String>builder()
-		.put("title", "Minecraft {mcversion}")
-		.build();
-	
-	private static final ImmutableSet<String> IGNORED_KEYS = ImmutableSet.of(
-		"icon16",
-		"icon32"
-	);
-	
+	private static final Map<String, String> DEFAULTS;
+	private static TitleConfig instance;
+
+	static {
+		final Map<String, String> defaults = new LinkedHashMap<>();
+		defaults.put("title", "Minecraft {mcversion}");
+		defaults.put("squareIcon", "");
+		DEFAULTS = Collections.unmodifiableMap(defaults);
+	}
+
 	public static TitleConfig read(final String folder) {
-		final Path configFile = Paths.get(folder, "customwindowtitle-client.toml");
-		final Map<String, String> config = new LinkedHashMap<>(DEFAULTS);
-		
-		try {
-			if (!Files.exists(configFile)) {
-				Files.write(configFile, config.entrySet().stream().map(entry -> String.format("%s = '%s'", entry.getKey(), entry.getValue())).collect(Collectors.toList()), StandardCharsets.UTF_8);
+		if (instance == null) {
```

Comment:
Lazy initialization is not thread-safe, all methods that access the field should be `synchronized`. Then the method should be renamed to `load`, and throw if called multiple times because that's not a supported operation.